### DISCUSSION TITLE
Inside Dockerfile, how to access url/branch most recently used in `dokku git:sync app giturl branch` ?

### DISCUSSION THREAD

**Discussion by pconrad** [dokku/dokku#6869] [Q&A] [answered]
I am deploying open source Spring Boot applications to dokku via this workflow:

```
dokku git:sync appname github-url branch
dokku ps:rebuild
```

Which is working fine, except for one thing.  I want to use [this plugin](https://github.com/git-commit-id/git-commit-id-maven-plugin) to get information about the most recent commit and expose it inside the application.  This is very helpful for knowing which version of the app is running right from the app itself.

It works in the dev environment (i.e. outside of dokku) where  the normal case is that the default directory where the final command to run the spring boot process is also a github repo directory.

<img width="1241" alt="image" src="https://github.com/dokku/dokku/assets/1119017/16498bee-cf22-4989-ab4a-95f41297d5a2">

But I've been unable to access any git information *inside the docker container*.   It does not appear to be the case that there is any directory that is a git directory, nor is there any environment variable that carries this information, and there doesn't seem to be any clear way to pass the information into the docker process automatically.   Since the spring-boot process isn't a git directory, you just get this:

<img width="501" alt="image" src="https://github.com/dokku/dokku/assets/1119017/d505423c-ee5f-4272-9bee-2959b94d494b">

Am I missing something? Or is what I'm trying to do simply not feasible?

Tl;dr: I want to be able, in the Dockerfile, to access the URL of the latest git sync and the name or SHA of the branch that was passed on the git:sync line so that I can make the

**Comment by pconrad** [comment]
Hmm... this seems related: https://github.com/dokku/dokku/discussions/5901  and give me some things to try, though it's not entirely clear whether it's a solution.

**Comment by josegonzalez** [comment, marked as answer]
I believe this was answered on Slack, but you can use the `keep-git-dir` git property: https://dokku.com/docs/deployment/methods/git/#keeping-the-git-directory

**Reply by pconrad** [reply]
Yes! Thanks for that.  I'm going to try it later today but for now I'm marking this as the answer.
